allennlp/common/checks.py

```diff
@@ -24,3 +24,12 @@ def __str__(self):
 def log_pytorch_version_info():
     import torch
     logger.info("Pytorch version: %s", torch.__version__)
+
+
+def check_dimensions_match(dimension_1: int,
+                           dimension_2: int,
+                           dim_1_name: str,
+                           dim_2_name: str) -> None:
```

Comment:
do you want to add an optional `message` field, so that we can include longer explanations when appropriate? (I see that this change currently eliminates some of them)

Reply:
I think the only place where there is information lost is in saying that the SRL model uses a binary verb indicator, which is actually incorrect (it was originally binary, it's now an embedding).  You still get the gist of (the correct part of) that information in the `"text embedding dim + verb indicator embedding dim"` message, so I'm not sure we're losing anything important.  If there's more to say, it should probably be said in the model docstring.